### PR TITLE
First methods

### PR DESCRIPTION
**Pull request by stephenlawrence**
Added the *First methods to the docs, renamed some deprecated methods to their non-deprecated names

### CODE REVIEW

**Review by mhald — APPROVED** [review]

File: docs/packaging-an-application/template-functions.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ url        = "/docs/packaging-an-application/template-functions"
 
 Template functions are marked by the double curly bracket + *"repl"* escape sequence. They allow for user input to be dynamically inserted into application configuration values. The sequence should be `{{repl`, not `{{ repl`.
 
-Template functions that refer to you containers are always addressed in pairs with "component name" and "image name".  You should use the full image name as it appears in your container definition. 
+Template functions that refer to you containers are always addressed in pairs with "component name" and "image name".  You should use the full image name as it appears in your container definition.
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: that refer to **your** containers

File: docs/packaging-an-application/template-functions.md [hunk 2]
```diff
@@ -74,64 +74,92 @@ ports:
      when: '{{repl ConfigOptionNotEquals "http_enabled" "1" }}'
 ```
 
-{{< template_function name="HostPrivateIpAddress" replicated="true" kubernetes="false" >}}
+{{< template_function name="NodePrivateIpAddress" replicated="true" kubernetes="false" >}}
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddress` <- case sensitive

File: docs/packaging-an-application/template-functions.md [hunk 3]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```go
-func HostPrivateIpAddress(componentName string, imageName string) string
+func NodePrivateIpAddress(componentName string, imageName string) string
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddress` <- case sensitive

File: docs/packaging-an-application/template-functions.md [hunk 4]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```
 Returns Private IP Address of Component as a string.
 
 ```yml
 env_vars:
 - name: REDIS_HOST_PRIVATE
-  static_val: '{{repl HostPrivateIpAddress "DB" "redis" }}'
+  static_val: '{{repl NodePrivateIpAddress "DB" "redis" }}'
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddress` <- case sensitive

File: docs/packaging-an-application/template-functions.md [hunk 5]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```
+Replaces HostPrivateIpAddress which is deprecated.
 
-{{< template_function name="HostPrivateIpAddressAll" replicated="true" kubernetes="false" >}}
+{{< template_function name="NodePrivateIpAddressFirst" replicated="true" kubernetes="false" >}}
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddressFirst` <- case sensitive

File: docs/packaging-an-application/template-functions.md [hunk 6]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```go
-func HostPrivateIpAddressAll(componentName string, imageName string) []string
+func NodePrivateIpAddressFirst(componentName string, imageName string) string
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddressFirst` <- case sensitive

File: docs/packaging-an-application/template-functions.md [hunk 7]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```
-Returns host private IP addresses for all instances of a given Component as an array of strings.
+Returns the first node's Private IP Address of Component as a string.
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `of a given Component`

File: docs/packaging-an-application/template-functions.md [hunk 8]
```diff
@@ -74,64 +74,92 @@ ports:
...
 
-Note: `ContainerExposedPortAll`, `HostPrivateIpAddressAll`, `HostPublicIpAddressAll` are guaranteed to return in the same order
+{{< template_function name="NodePrivateIpAddressAll" replicated="true" kubernetes="false" >}}
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 9]
```diff
@@ -74,64 +74,92 @@ ports:
...
+```go
+func NodePrivateIpAddressAll(componentName string, imageName string) []string
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 10]
```diff
@@ -74,64 +74,92 @@ ports:
...
+```
+Returns node private IP addresses for all instances of a given Component as an array of strings.
+Replaces HostPrivateIpAddressAll which is deprecated.
 
-{{< template_function name="HostPublicIpAddress" replicated="true" kubernetes="false" >}}
+Note: `ContainerExposedPortAll`, `NodePrivateIpAddressAll`, `NodePublicIpAddressAll` are guaranteed to return in the same order
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddressAll` `NodePublicIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 11]
```diff
@@ -74,64 +74,92 @@ ports:
...
+
+{{< template_function name="NodePublicIpAddress" replicated="true" kubernetes="false" >}}
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePublicIPAddress`

File: docs/packaging-an-application/template-functions.md [hunk 12]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```go
-func HostPublicIpAddress(componentName string, imageName string) string
+func NodePublicIpAddress(componentName string, imageName string) string
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePublicIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 13]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```
 Returns Public IP Address of Component as a string.
 ```yml
 env_vars:
 - name: REDIS_HOST_PUBLIC
-  static_val: '{{repl HostPublicIpAddress "DB" "redis" }}'
+  static_val: '{{repl NodePublicIpAddress "DB" "redis" }}'
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePublicIPAddress`

File: docs/packaging-an-application/template-functions.md [hunk 14]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```
+Replaces HostPublicIpAddress which is deprecated.
 
-{{< template_function name="HostPublicIpAddressAll" replicated="true" kubernetes="false" >}}
+{{< template_function name="NodePublicIpAddressFirst" replicated="true" kubernetes="false" >}}
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePublicIPAddressFirst`

File: docs/packaging-an-application/template-functions.md [hunk 15]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```go
-func HostPublicIpAddressAll(componentName string, imageName string) []string
+func NodePublicIpAddressFirst(componentName string, imageName string) string
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePublicIPAddressFirst`

File: docs/packaging-an-application/template-functions.md (hunk 10 above)
Comment (emosbaugh) [reply docs/packaging-an-application/template-functions.md]: I'm unsure about this statement but it should be true.

File: docs/packaging-an-application/template-functions.md [hunk 16]
```diff
@@ -74,64 +74,92 @@ ports:
...
+```
+Returns first node's public IP addresses for a given Component as a string.
+
+{{< template_function name="NodePublicIpAddressAll" replicated="true" kubernetes="false" >}}
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePublicIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 17]
```diff
@@ -74,64 +74,92 @@ ports:
...
+```go
+func NodePublicIpAddressAll(componentName string, imageName string) []string
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePublicIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 18]
```diff
@@ -74,64 +74,92 @@ ports:
...
 ```
 Returns host public IP addresses for all instances of a given Component as an array of strings.
+Replaces HostPublicIpAddressAll which is deprecated.
 
-Note: `ContainerExposedPortAll`, `HostPrivateIpAddressAll`, `HostPublicIpAddressAll` are guaranteed to return in the same order
+Note: `ContainerExposedPortAll`, `NodePrivateIpAddressAll`, `Node PublicIpAddressAll` are guaranteed to return in the same order
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddressAll` `NodePublicIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 19]
```diff
@@ -74,64 +74,92 @@ ports:
...
 
 ```yml
 env_vars:
 - name: REDIS_PORT
   static_val: '{{repl ContainerExposedPort "DB" "redis" "6379" }}'
 ```
 
+{{< template_function name="ContainerExposedPortFirst" replicated="true" kubernetes="false" >}}
+```go
+func ContainerExposedPortFirst(componentName string, imagename string, internalPort string) string
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `imageName`

File: docs/packaging-an-application/template-functions.md [hunk 20]
```diff
@@ -74,64 +74,92 @@ ports:
...
+```
+
 {{< template_function name="ContainerExposedPortAll" replicated="true" kubernetes="false" >}}
 ```go
 func ContainerExposedPortAll(componentName string, imageName string, internalPort string) string
 ```
 Returns the host public port mapped to the supplied exposed container port for all instances of a given Component as an array of strings.
 
-Note: `ContainerExposedPortAll`, `HostPrivateIpAddressAll`, `HostPublicIpAddressAll` are guaranteed to return in the same order
+Note: `ContainerExposedPortAll`, `NodePrivateIpAddressAll`, `NodePublicIpAddressAll` are guaranteed to return in the same order
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: `NodePrivateIPAddressAll` `NodePublicIPAddressAll`

File: docs/packaging-an-application/template-functions.md [hunk 21]
```diff
@@ -74,64 +74,92 @@ ports:
...
-func HostPublicIpAddressAll(componentName string, imageName string) []string
+func NodePublicIpAddressFirst(componentName string, imageName string) string
+```
+Returns first node's public IP addresses for a given Component as a string.
+
+{{< template_function name="NodePublicIpAddressAll" replicated="true" kubernetes="false" >}}
+```go
+func NodePublicIpAddressAll(componentName string, imageName string) []string
 ```
 Returns host public IP addresses for all instances of a given Component as an array of strings.
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: Returns **node** public IP addresses

File: docs/packaging-an-application/template-functions.md [hunk 22]
```diff
@@ -74,64 +74,92 @@ ports:
...
+env_vars:
+- name: REDIS_PORT
+  static_val: '{{repl ContainerExposedPortFirst "DB" "redis" "6379" }}'
+```
+
 {{< template_function name="ContainerExposedPortAll" replicated="true" kubernetes="false" >}}
 ```go
 func ContainerExposedPortAll(componentName string, imageName string, internalPort string) string
 ```
 Returns the host public port mapped to the supplied exposed container port for all instances of a given Component as an array of strings.
```
Comment (emosbaugh) [inline docs/packaging-an-application/template-functions.md]: Returns the **node** public

**Review by emosbaugh — CHANGES_REQUESTED** [review]

**Comment by stephenlawrence** [comment]
Updated with PR fixes. @emosbaugh

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**